cloudinit/config/cloud-init-schema.json

```diff
@@ -543,6 +543,244 @@
           }
         }
       }
+    },
+    "cc_package_update_upgrade_install": {
+        "type": "object",
+        "properties": {
+          "packages": {
+            "type": "array",
+            "description": "A list of packages to install. Each entry in the list can be either a package name or a list with two entries, the first being the package name and the second being the specific package version to install.",
+              "items": {
+                "oneOf": [
+                  {"type": "array", "items": {"type": "string"}, "minItems": 2, "maxItems": 2},
+                  {"type": "string"}
+                ]
+              },
+              "minItems": 1
+            },
+            "package_update": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to update packages. Happens before upgrade or install. Default: ``false``"
+            },
+            "package_upgrade": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to upgrade packages. Happens before install. Default: ``false``"
+            },
+            "package_reboot_if_required": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to reboot the system if required. Default: ``false``"
+            },
+            "apt_update": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_update``. Default: ``false``"
+            },
+            "apt_upgrade": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_upgrade``. Default: ``false``"
+            },
+            "apt_reboot_if_required": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_reboot_if_required``. Default: ``false``"
+            }
+        }
+    },
+    "cc_phone_home": {
+      "type": "object",
+      "properties": {
+        "phone_home": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["url"],
+          "properties": {
+            "url": {
```

Comment:
Since jsonschema has had a [builtin uri format going back to Draft 4](https://json-schema.org/understanding-json-schema/reference/string.html#format), this might be a good location to see if it can be used for validation of the URL here? I'm also not sure if it's worth the effort.